scripts/etc/init.d/rcS

```diff
@@ -217,19 +232,26 @@ for i in `seq 1 10`; do
     echo -n "$i.. "
 done
 
+if [ "$ifname" != "eth0" ]; then
+    # if eth0 is not the chosen eth interface it is a wireless interface and wpa_supplicant must connect to wlan
+    echo -n "Running wpa_supplicant... "
+    wpa_supplicant -B -Dwext -c/bootfs/config/wpa_supplicant.conf -iwlan0 >& /dev/null || fail
```

Comment:
I'd change `-Dwext` to `-Dnl80211,wext`. `nl80211` is the 'new' wireless standard so I'd very much like to support that. `wext` is deprecated, but without it a lot of wireless cards probably won't work.
The `-iwlan0` should probably be `-i$ifname`